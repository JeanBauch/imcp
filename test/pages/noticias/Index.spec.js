import { mount } from '@vue/test-utils';

import Index from '~/pages/noticias/index.vue';

describe('Pages:index', () => {
	let mountConfig;

	beforeEach(() => {
		mountConfig = {
			stubs: {
				Breadcrumb: true,
				TitlePage: true,
				CardNoticia: true,
			},
		};
	});

	describe('Snapshots', () => {
		it('Should render all child componets', () => {
			const wrapper = mount(Index, mountConfig);
			expect(wrapper.html()).toMatchSnapshot();
		});
	});
});
