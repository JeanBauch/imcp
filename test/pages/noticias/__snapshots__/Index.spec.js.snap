// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pages:index Snapshots Should render all child componets 1`] = `
"<div class=\\"bg-background\\">
  <section class=\\"container mx-auto px-6 sm:px-9 lg:px-24\\">
    <div class=\\"flex flex-1 py-3\\">
      <breadcrumb-stub crumbs=\\"[object Object],[object Object]\\"></breadcrumb-stub>
    </div>
    <div class=\\"flex flex-1 items-center justify-center mt-4 md:mt-6\\">
      <titlepage-stub title=\\"NotÃ­cias\\" titlecolor=\\"orange\\" separetorcolor=\\"yellow\\" width=\\"w-min\\"></titlepage-stub>
    </div>
  </section>
  <main class=\\"container mx-auto px-6 sm:px-9 lg:px-24 mt-6 md:mt-8 lg:mt-10 mb-56\\">
    <div class=\\"flex flex-col justify-center gap-8\\">
      <div class=\\"flex flex-col md:flex-row items-center md:justify-between rounded-3xl bg-primary-bordo-500 p-8 gap-3\\">
        <h3 class=\\"text-white text-base font-medium uppercase block md:hidden\\">
          Filtrar
        </h3>
        <div class=\\"w-full md:w-[70%] xl:w-1/3 flex items-center gap-4\\">
          <h6 class=\\"text-white text-sm xl:text-base hidden md:block\\">
            Categorias:
          </h6>
          <div class=\\"w-full border border-white rounded-full px-6 py-3\\">
            <div class=\\"flex items-center h-5\\">
              <h5 class=\\"text-white text-sm font-semibold\\">
                Todos
              </h5>
            </div>
          </div>
        </div>
        <div class=\\"w-full xl:w-5/12 flex items-center gap-4\\">
          <h6 class=\\"text-white text-xs xl:text-base hidden md:block\\">
            Buscar por palavra-chave:
          </h6>
          <div class=\\"flex flex-1 border border-white rounded-full px-6 py-3\\">
            <div class=\\"flex items-center h-5\\">
              <h5 class=\\"text-white text-xs\\">
                Ex: Coral das Meninas
              </h5>
            </div>
          </div>
        </div>
      </div>
      <div class=\\"grid grid-cols-1 sm:grid-cols-2 lg:grid-cols-6 gap-8 transition-all\\">
        <cardnoticia-stub islarger=\\"true\\" class=\\"md:col-span-1 lg:col-span-3\\"></cardnoticia-stub>
        <cardnoticia-stub islarger=\\"true\\" class=\\"md:col-span-1 lg:col-span-3\\"></cardnoticia-stub>
        <cardnoticia-stub class=\\"md:col-span-1 lg:col-span-2\\"></cardnoticia-stub>
        <cardnoticia-stub class=\\"md:col-span-1 lg:col-span-2\\"></cardnoticia-stub>
        <cardnoticia-stub class=\\"md:col-span-1 lg:col-span-2\\"></cardnoticia-stub>
        <cardnoticia-stub islarger=\\"true\\" class=\\"md:col-span-1 lg:col-span-3\\"></cardnoticia-stub>
        <cardnoticia-stub islarger=\\"true\\" class=\\"md:col-span-1 lg:col-span-3\\"></cardnoticia-stub>
      </div>
      <div class=\\"flex flex-1 items-center justify-center gap-4 mt-4 text-primary-bordo-500\\"><button><svg xmlns=\\"http://www.w3.org/2000/svg\\" fill=\\"none\\" viewBox=\\"0 0 24 24\\" stroke=\\"currentColor\\" stroke-width=\\"2\\" class=\\"h-4 w-4\\">
            <path stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\" d=\\"M15 19l-7-7 7-7\\"></path>
          </svg></button> <button class=\\"bg-primary-bordo-500 w-8 h-8 rounded-full text-white\\">
          1
        </button> <button>2</button> <button>3</button>
        <p>...</p> <button>10</button> <button><svg xmlns=\\"http://www.w3.org/2000/svg\\" fill=\\"none\\" viewBox=\\"0 0 24 24\\" stroke=\\"currentColor\\" stroke-width=\\"2\\" class=\\"h-4 w-4\\">
            <path stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\" d=\\"M9 5l7 7-7 7\\"></path>
          </svg></button>
      </div>
    </div>
  </main>
</div>"
`;
